modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -217,41 +222,50 @@ object Inputs {
     cwd: os.Path,
     directories: Directories,
     baseProjectName: String,
-    download: String => Either[String, Array[Byte]],
+    download: String => Either[String, List[UrlPath]],
```

Comment:
I think we should need to change that, and add `UrlPath`, here. The `Array[Byte]` could be changed to `os.Path` if needed, but that would only be a nice-to-have (see my comment below).